src/mam/mod_mam_rdbms_arch.erl

```diff
@@ -341,15 +341,18 @@ remove_archive(Acc, HostType, ArcID, _ArcJID) ->
     mongoose_rdbms:execute_successfully(HostType, mam_archive_remove, [ArcID]),
     Acc.
 
--spec remove_domain(mongoose_hooks:simple_acc(), host_type(), jid:lserver()) ->
-    mongoose_hooks:simple_acc().
+-spec remove_domain(mongoose_domain_api:remove_domain_acc(), host_type(), jid:lserver()) ->
+    mongoose_domain_api:remove_domain_acc().
 remove_domain(Acc, HostType, Domain) ->
-    {atomic, _} = mongoose_rdbms:sql_transaction(HostType, fun() ->
-            mongoose_rdbms:execute_successfully(HostType, mam_remove_domain, [Domain]),
-            mongoose_rdbms:execute_successfully(HostType, mam_remove_domain_prefs, [Domain]),
-            mongoose_rdbms:execute_successfully(HostType, mam_remove_domain_users, [Domain])
-        end),
-    Acc.
+    F = fun() ->
```

Comment:
It is really unexpected that this function is (and most likely was already) untouched by tests. Could you investigate? I thought we had tests for domain removal for MAM.

Reply:
No idea, I just ran locally with coverage and it is green in the report generated here, I don't know why it wasn't for codecov 😕

Reply:
codecov says that for master is also red: https://app.codecov.io/gh/esl/MongooseIM/blob/master/src/mam/mod_mam_rdbms_arch.erl so well, I'm sure we have a test for it, in https://github.com/esl/MongooseIM/blob/e79f08474226283c5de226ce7aeea2734a8112f1/big_tests/tests/domain_removal_SUITE.erl#L167, so it seems like something's off with codecov 😅

Reply:
Ok, let's leave it for now